posts/2024-05-13-jakarta-nosql-challenge-switching-nosql-easily.adoc

```diff
@@ -0,0 +1,258 @@
+// can be left empty which will set it to the default
+//
+// "open-graph-image-alt" is a description of what is in the image (not a caption). When changing "open-graph-image" to
+// a custom picture, you must provide a custom string for "open-graph-image-alt".
+//
+// Replace DESCRIPTION with a short summary (~60 words) of the release (a more succinct version of the first paragraph of the post).
+//
+// If adding image into the post add :
+// -------------------------
+// [.img_border_light]
+// image::img/blog/FILE_NAME[IMAGE CAPTION ,width=70%,align="center"]
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around screenshots but not
+// around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+Nowadays, in the cloud era, switching between NoSQL solutions could be interesting to save resources and costs, but it could be a nightmare, specially if we're highly coupled to a particular vendor. It will cause some changes that will make this approach very expensive, like: time spent on the change, the learning curve of a new API to use with this database, the code that will be lost, the persistence layer that needs to be replaced, etc.
+
+This is where the flexibility of Jakarta NoSQL with an implementation like https://www.jnosql.org[Eclipse JNoSQL] together shines.
+
+NOTE: As of March 2024, Eclipse JNoSQL supports about https://www.jnosql.org/docs/supported_dbs.html[30 NoSQL databases].
+
+== Jakarta NoSQL in Action: Switching NoSQL Databases with Ease
+
+To explore this capability, let's use the link:../../../2024/05/06/jakarta-nosql-in-action-meet-jnopo-game.html[*JNopo* game].
+
+Here's the challenge: let's suppose that it's needed to change the *JNopo*'s Document NoSQL database, *MongoDB*, to another Document NoSQL database for save money purposes, e.g. *Couchbase*.
+
+By using Jakarta NoSQL specification to connect to the NoSQL solutions, we got some benefits and one of them is that with just a few changes, we'll be able to achive this goal easily.
+
+=== Configuring Eclipse JNoSQL dependencies for Couchbase
+
+In the `pom.xml` file, add the following dependency:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.eclipse.jnosql.databases</groupId>
+  <artifactId>jnosql-couchbase</artifactId>
+  <version>1.1.0</version>
+</dependency>
+----
+
+=== Configuring a Local Couchbase Cluster
+
+First of all, let's add the following content to the `docker-compose.yml` file:
```

Comment:
```suggestion
To successfully configure our new Couchbase cluster, we'll first need to add the following content to the `docker-compose.yml` file:
```

Reply:
Would be nice to have a short sentence here explaining why we have done this. At the moment it feels a little like we're getting the audience to just follow steps without giving them the explanation/context as to why these steps are needed so they can apply it to their own applications.